spynnaker/pyNN/utilities/database/data_base_interface.py

```diff
@@ -0,0 +1,564 @@
+            "CREATE TABLE graph_mapper_edges("
+            "partitionable_edge_id INTEGER, partitioned_edge_id INTEGER, "
+            "PRIMARY KEY(partitionable_edge_id, partitioned_edge_id), "
+            "FOREIGN KEY (partitioned_edge_id)"
+            " REFERENCES Partitioned_edges(edge_id), "
+            "FOREIGN KEY (partitionable_edge_id)"
+            " REFERENCES Partitionable_edges(edge_id))")
+        cur.execute(
+            "CREATE TABLE Placements("
+            "vertex_id INTEGER PRIMARY KEY, machine_id INTEGER, "
+            "chip_x INT, chip_y INT, chip_p INT, "
+            "FOREIGN KEY (vertex_id) "
+            "REFERENCES Partitioned_vertices(vertex_id), "
+            "FOREIGN KEY (chip_x, chip_y, chip_p, machine_id) "
+            "REFERENCES Processor(chip_x, chip_y, physical_id, "
+            "machine_id))")
+        cur.execute(
+            "CREATE TABLE Routing_info("
+            "edge_id INTEGER PRIMARY KEY, key INT, mask INT, "
+            "FOREIGN KEY (edge_id) REFERENCES Partitioned_edges(edge_id))")
+        cur.execute(
+            "CREATE TABLE Routing_table("
+            "chip_x INTEGER, chip_y INTEGER, position INTEGER, "
+            "key_combo INT, mask INT, route INT, "
+            "PRIMARY KEY (chip_x, chip_y, position))")
+        cur.execute(
+            "CREATE TABLE configuration_parameters("
+            "parameter_id TEXT, value REAL, "
+            "PRIMARY KEY (parameter_id))")
+
+    # noinspection PyPep8
+    def send_visualiser_notifcation(self):
+        self._wait_pool.apply_async(self._send_visualiser_notifcation)
+
+    def _send_visualiser_notifcation(self):
+        try:
+            self._sent_visualisation_confirmation = True
+            self._thread_pool.close()
+            self._thread_pool.join()
+            #after all writing, send notifcation to vis
+            if self._wait_for_vis:
+                self._notify_visualiser_and_wait()
+        except Exception as e:
+            print e
+
+    def _notify_visualiser_and_wait(self):
+
+        data_base_message_connections = list()
+        for socket_address in self._socket_addresses:
+            data_base_message_connection = \
+                EieioCommandConnection(
+                    socket_address.listen_port, socket_address.notify_host_name,
+                    socket_address.notify_port_no)
+            data_base_message_connections.append(data_base_message_connection)
+
+        # create complete message for vis to pick up
+        eieio_command_header = EIEIOCommandHeader(
+            spinnman_constants.EIEIO_COMMAND_IDS.DATABASE_CONFIRMATION.value)
+        eieio_command_message = EIEIOCommandMessage(eieio_command_header)
+        # add file path to database into command message.
+        #|------P------||------F-----|---------path----------|
+        #       0              1               path
+        send_file_path = config.getboolean("Database", "send_file_path")
+        if send_file_path:
```

Comment:
Not sure that we want to make this optional - is there any event in which you don't want it?